File: rreadr_script.jl
using RCall

R"""
memory.limit(size=32000)
library(readr)
options(readr.num_columns = 0)
""";

warmup_filename = ARGS[1]
filename = ARGS[2]

t1 = convert(Float64, R"""
    start_time = Sys.time()
    suppressMessages(read_csv($warmup_filename, progress = FALSE))
    end_time = Sys.time()
    elap = end_time - start_time
""")

GC.gc(); GC.gc(); GC.gc()

t2 = convert(Float64, R"""
    start_time = Sys.time()
    suppressMessages(read_csv($filename, progress = FALSE))
    end_time = Sys.time()
    elap = end_time - start_time
""")

GC.gc(); GC.gc(); GC.gc()

t3 = convert(Float64, R"""
    start_time = Sys.time()
    suppressMessages(read_csv($filename, progress = FALSE))
    end_time = Sys.time()
    elap = end_time - start_time
""")

println(t1)
println(t2)
println(t3)

File: src/packagescripts/rreadr_script.R
library(readr)
options(readr.num_columns = 0)

args = commandArgs(trailingOnly=TRUE)

warmup_filename = args[1]
filename = args[2]

start_time = Sys.time()
x1 <- read_csv(warmup_filename, progress = FALSE)
end_time = Sys.time()
t1 = end_time - start_time


start_time = Sys.time()
x2 <- read_csv(filename, progress = FALSE)
end_time = Sys.time()
t2 = end_time - start_time


start_time = Sys.time()
x3 <- read_csv(filename, progress = FALSE)
end_time = Sys.time()
t3 = end_time - start_time

cat(as.numeric(t1))
cat('\n')
cat(t2)
cat('\n')
cat(t3)
cat('\n')
